0_portfolio_input_check_functions.R

```diff
@@ -1538,16 +1539,37 @@ add_other_to_sector_classifications <- function(audit) {
 }
 
 
-# add_bics_sector <- function(portfolio){
-#   #join in bics sectors for EQ and CB via bics_bridge
-#   bics_bridge <- read_csv("data/bics_bridge.csv")
-#   portfolio <- portfolio %>%
-#     left_join(bics_bridge, by = c("security_bics_subgroup" = "bics_subgroup"))
-#
-#   if ("bics_sector.x" %in% colnames(portfolio)) {
-#     portfolio <- portfolio %>% select(-bics_sector.x) %>% rename(bics_sector = bics_sector.y)
-#   }
-#
-# return(portfolio)
-#
-# }
+pw_calculations <- function(eq_portfolio, cb_portfolio){
+
+  port_all <- data.frame()
+
+  if(data_check(eq_portfolio)){
+
+    port_all <- bind_rows(port_all, eq_portfolio)
+  }
+
+  if(data_check(cb_portfolio)){
+
+    port_all <- bind_rows(port_all, cb_portfolio)
+
+  }
+
+  if(data_check(port_all)){
+
+  port_all <- port_all %>%  select(grouping_variables,company_id, value_usd)
+
+  port_all <- calculate_port_weight(port_all, grouping_variables)
+
+
+  pw <- port_all %>%
+    group_by(!!!rlang::syms(grouping_variables), company_id) %>%
```

Comment:
so yeah, this seems more appropriate way to use `grouping_variables`.... though I'm not sure if the syntax here is the most appropriate/efficient

Reply:
Yeah this is what was working a while ago re syntax so copied that. can change at some point if there is something better.